evalml/automl/automl_algorithm/iterative_algorithm.py

```diff
@@ -129,6 +196,141 @@ def __init__(
+                and self._pipeline_params["pipeline"]["date_index"]
+            ):
+                if (
+                    pd.infer_freq(
+                        self.X[self._pipeline_params["pipeline"]["date_index"]]
+                    )
+                    == "MS"
+                ):
+                    allowed_estimators = [
+                        estimator
+                        for estimator in allowed_estimators
+                        if estimator.name != "ARIMA Regressor"
+                    ]
+            self.logger.debug(
+                f"allowed_estimators set to {[estimator.name for estimator in allowed_estimators]}"
+            )
+            drop_columns = (
+                self._pipeline_params["Drop Columns Transformer"]["columns"]
+                if "Drop Columns Transformer" in self._pipeline_params
+                else None
+            )
+            index_and_unknown_columns = list(
+                self.X.ww.select(["index", "unknown"], return_schema=True).columns
+            )
+            unknown_columns = list(
+                self.X.ww.select("unknown", return_schema=True).columns
+            )
+            index_and_unknown_columns = index_and_unknown_columns
+            if len(index_and_unknown_columns) > 0 and drop_columns is None:
+                self._pipeline_params["Drop Columns Transformer"] = {
+                    "columns": index_and_unknown_columns
+                }
+                if len(unknown_columns):
+                    self.logger.info(
+                        f"Removing columns {unknown_columns} because they are of 'Unknown' type"
+                    )
+
+            with warnings.catch_warnings(record=True) as w:
+                warnings.filterwarnings("always", category=ParameterNotUsedWarning)
+                self.allowed_pipelines = [
+                    make_pipeline(
+                        self.X,
+                        self.y,
+                        estimator,
+                        self.problem_type,
+                        parameters=self._pipeline_params,
+                        sampler_name=self.sampler_name,
+                    )
+                    for estimator in allowed_estimators
+                ]
+            self._catch_warnings(w)
+        else:
+            with warnings.catch_warnings(record=True) as w:
+                warnings.filterwarnings("always", category=ParameterNotUsedWarning)
+                self.allowed_pipelines = get_pipelines_from_component_graphs(
+                    self.allowed_component_graphs,
+                    self.problem_type,
+                    self._pipeline_params,
+                    self.random_seed,
+                )
+            self._catch_warnings(w)
+
+        if self.allowed_pipelines == []:
+            raise ValueError("No allowed pipelines to search")
+
+        if self.ensembling and len(self.allowed_pipelines) == 1:
+            self.logger.warning(
```

Comment:
I also opted to move all the logging in as well. From my understanding of the logger, there will be no change in output.